### PR TITLE
chore(build): Migrate to Develocity build cache connector

### LINKED ISSUE #105: Migrate to Develocity build cache connector
Apply Change similar to https://github.com/grails/grails-core/pull/13387

### PR DESCRIPTION
**Pull request by guillermocalvo**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -62,8 +58,6 @@ jobs:
         uses: micronaut-projects/github-pages-deploy-action@grails
         env:
```
Comment (matrei) [inline .github/workflows/release.yml:59]: The micronaut-projects/github-pages-deploy-action does not really use gradle so the whole env block can be removed.

Comment (guillermocalvo) [reply .github/workflows/release.yml:59]: @matrei Cool, thanks! What about `TARGET_REPOSITORY`, `GH_TOKEN`, and `BRANCH`? Aren't these used?

Comment (matrei) [reply .github/workflows/release.yml:59]: Yes, sorry, that was wrong to say the whole env block could be removed. Only the GRADLE_ENTERPRISE_* vars can be removed.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -62,8 +58,6 @@ jobs:
...
           GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
```
Comment (guillermocalvo) [inline .github/workflows/release.yml]:
```suggestion
```

File: .github/workflows/release.yml (hunk 1 above)
Comment (guillermocalvo) [reply .github/workflows/release.yml:59]: @matrei No worries 🙂 It's good to know that we can drop all those GRADLE_ENTERPRISE_* env vars. Thank you!

**Review by matrei — APPROVED** [review]

**Pull request merged**